subprojects/core-api/src/main/java/org/gradle/api/file/ConfigurableFileCollection.java

```diff
@@ -42,14 +42,14 @@ public interface ConfigurableFileCollection extends FileCollection, HasConfigura
     /**
      * Sets the source paths for this collection. The given paths are evaluated as per {@link org.gradle.api.Project#files(Object...)}.
      *
-     * @param paths The paths.
+     * @param paths The paths. {@code null} values are ignored.
```

Comment:
🤔  There is also a misalignment with the types. The package is non-null-annotated, so these arguments are never supposed to be null. If we were good citizens, we would have actually checked the argument for null and were still throwing NPE, but early.

So, when we figure out what to do with nulls in CFC, it'd be nice to have this consistent with the argument nullability of the API as well.